File: plugins/calculate.py
import re
from plugin import Plugin


class CalculatePlugin(Plugin):
    def translate(self, line):
        match = re.fullmatch(
            r'計算する\s+(.+)',
            line
        )

        if not match:
            return None

        expression = match.group(1)

        return f"print({expression})"

File: web/plugins/calculate.js
import { WebPlugin } from "../engine/plugin.js";

export class CalculatePlugin extends WebPlugin {
    constructor() {
        super("計算する");
    }

    async execute(line, context) {
        const expression = line
            .replace(/^計算する\s*/, "");

        const value = context.parser.evaluate(
            expression,
            context.runtime
        );

        context.runtime.write(value);
    }
}